src/Servers/HttpSys/src/RequestProcessing/RequestHeaders.Generated.tt

```diff
@@ -213,5 +212,36 @@ namespace Microsoft.AspNetCore.HttpSys.Internal
             }
 <# } #>
         }
+
+        private static ReadOnlySpan<byte> HeaderKeys => new byte[]
+        {
+<# foreach(var prop in props) { #>
+            (byte)<#=prop.ID#>,
+<# } #>
+        };
+
+        private string GetHeaderKeyName(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => HeaderNames.<#=prop.Name#>,
+<# } #>
+            _ => throw new NotImplementedException()
+        };
+
+        private bool IsRead(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => <#=IsRead(prop.Index)#>,
+<# } #>
+            _ => throw new NotImplementedException()
+        };
+
+        private bool IsKnownHeaderSet(HttpSysRequestHeader header) => header switch
```

Comment:
Maybe `IsKnownHeaderPresent` is a better name?

Reply:
I still like HasKnownHeader, but that's ambiguous with the other new one. Oh, you don't need the other one anymore, it's only called here. You can inline the `_requestMemoryBlob.HasKnownHeader(header)` call and get rid of the other method.

Reply:
ack